erpnext/controllers/selling_controller.py

```diff
@@ -416,6 +419,17 @@ def check_stop_sales_order(self, ref_fieldname):
 				status = frappe.db.get_value("Sales Order", d.get(ref_fieldname), "status")
 				if status == "Stopped":
 					frappe.throw(_("Sales Order {0} is stopped").format(d.get(ref_fieldname)))
+	
+	def is_item_table_empty(self):
+		if not len(self.get(self.fname)):
+			frappe.throw(_("Item table can not be blank"))
+					
+	def set_qty_as_per_stock_uom(self):
+		for d in self.get(self.fname):
+			if d.meta.get_field("stock_qty") and not d.stock_qty:
+				if not d.conversion_factor:
+					frappe.throw(_("Item {0}: Conversion Factor is mandatory").format(d.item_code))
+				d.stock_qty = flt(d.qty) * flt(d.conversion_factor)
```

Comment:
Set stock_qty as per precision of the field:

`flt(flt(d.qty) * flt(d.conversion_factor), self.precision("stock_qty"))`